envelop/__init__.py

```diff
@@ -17,9 +17,10 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 from __future__ import unicode_literals
+from six.moves.urllib.parse import urlparse
+from builtins import str as text
```

Comment:
Unless we use `futures` directly, I wouldn't do this because we would need to install a library we don't use and all of its dependencies (in the future they may change the lib to have extra dependencies). If we're already depending on `six` and it provides the type we want, I'd use it; otherwise define a small try/except for the type. I guess six implements `text_type` as

```python
try:
    from builtins import str as text_type
except ImportError:
    text_type = unicode
```